### ISSUE TITLE
Plugin API: query selector

### ISSUE THREAD

**Issue by joscha** [pecuniabanking/pecunia-client#430]
Hi,

when running this code in a custom plugin:
```javascript
var doc = webClient.mainFrame.document;

var nativeForms = doc.forms;
logger.logVerbose(nativeForms.length);

var forms = doc.querySelectorAll('form');
logger.logVerbose(forms.length);

logger.logVerbose(webClient.stringByEvaluatingJavaScriptFromString("document.querySelectorAll('form').length"));
```
The output surprisingly is:
```
[Verbose] 1
[Verbose] 0
[Verbose] 0
```

I looked into the code of the plugin bridge but couldn't find anything that sticks out why the behaviour would be so drastically different. Any hints?

**Comment by joscha** [comment]
If I accessed the respective Swift API once before like so right after `(*) Finished loading frame from URL`:
```swift
        let links = webClient.mainFrame.document.querySelectorAll("[title*='Login']")
        jsLogger.logVerbose(String(links.length));
        var i : UInt32;
        for (i = 0; i < links.length; i++) {
            jsLogger.logVerbose(links.item(i).stringRepresentation());
        }
```
it works. There must be some sort of mutation state that only initializes the respective JS API when the according Swift code has been executed before.

**Comment by mike-lischke** [comment]
Well, that must be something internal to Webkit. I'm not aware of anything I influence there in this regard.

**Comment by joscha** [comment]
Yes, it's very strange - I'd like to isolate the problem: do we have tests somewhere for the Plugin API?

**Comment by mike-lischke** [comment]
Do you mean for the Pecunia Plugin API or Webkit? I don't know any tests.

**Comment by joscha** [comment]
For the Pecunia Plugin API. Does Pecunia itself have any tests?

**Comment by joscha** [comment]
I am kind of stuck with this. Tried for hours to work around it and find out where it breaks. When being used within Swift, everything works like a charm. When being used in Swift before using it in JS, it works as well (parameters have to be the same, e.g. `document.querySelector('x')` in Swift would make `document.querySelector('x')` in JS work, but not `document.querySelector('y')`. When being used in JavaScript without using it prior in Swift, an empty list is returned. It is almost as if there is some sort of caching layer in between the JS interface and the Swift interface.

**Comment by mike-lischke** [comment]
If all fails we may have to add a special querySelector() function to the webView Swift class, to be used instead of the one from JS. Btw, I had a similar issue in the DKB plugin, not finding a form by name, so I iterated over all forms and compared names manually: https://github.com/pecuniabanking/pecunia-client/blob/master/Plugins/dkb.js#L256.
